tests/hash_table_test.c

```diff
@@ -1258,3 +1258,39 @@ static int s_test_hash_table_byte_cursor_create_find_fn(struct aws_allocator *al
 
     return 0;
 }
+
+AWS_TEST_CASE(test_hash_combine, s_test_hash_combine_fn)
+static int s_test_hash_combine_fn(struct aws_allocator *allocator, void *ctx) {
+    uint64_t a, b, c;
+
+    /* We're assuming that the underlying hashing function works,
+     * just making sure we hooked it up right for 2 64bit values */
+
+    a = 0x123456789abcdef;
+    b = 0xfedcba987654321;
+    c = aws_hash_combine(a, b);
+
+    /* Sanity check */
+    ASSERT_TRUE(c != a);
+    ASSERT_TRUE(c != b);
+
+    /* Same inputs gets same results, right? */
+    ASSERT_UINT_EQUALS(c, aws_hash_combine(a, b));
+
+    /* Using all bytes, right? */
+    uint8_t *c_bytes = (uint8_t *)&c;
+    for (size_t i = 0; i < sizeof(c); ++i) {
+        ASSERT_TRUE(c_bytes[i] != 0);
+    }
+
+    /* Check even low-entropy numbers result in something with high entropy */
```

Comment:
If we wanted to test this, wouldn't we just fuzz it?

Reply:
I just took this part of the test out. This is not the place to test the quality of the underlying hash library